x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/CO2Calculator.java

```diff
@@ -7,212 +7,194 @@
-        return provider.PUE;
+        return PROVIDERS.getOrDefault(host.instanceType.provider, DEFAULT_PROVIDER).pue;
     }
 
-    private static class Provider {
-        double PUE;
-        Map<String, Double> co2TonsPerKWH;
-
-        Provider(double PUE, Map<String, Double> co2TonsPerKWH) {
-            this.PUE = PUE;
-            this.co2TonsPerKWH = co2TonsPerKWH;
-        }
-    }
+    private record Provider(double pue, Map<String, Double> co2TonsPerKWH) {}
 
     // values are taken from https://www.cloudcarbonfootprint.org/docs/methodology/
-    private static final Map<String, Provider> providers = new HashMap<>();
+    private static final Map<String, Provider> PROVIDERS;
     static {
         // noinspection (explicit type arguments speedup compilation and analysis time)
-        providers.put(
+        PROVIDERS = Map.of(
             "aws",
             new Provider(
                 1.135d,
                 Map.ofEntries(
-                    new AbstractMap.SimpleEntry<>("us-east-1", 0.000379069d),
-                    new AbstractMap.SimpleEntry<>("us-east-2", 0.000410608d),
-                    new AbstractMap.SimpleEntry<>("us-west-1", 0.000322167d),
-                    new AbstractMap.SimpleEntry<>("us-west-2", 0.000322167d),
-                    new AbstractMap.SimpleEntry<>("us-gov-east-1", 0.000379069d),
-                    new AbstractMap.SimpleEntry<>("us-gov-west-1", 0.000322167d),
-                    new AbstractMap.SimpleEntry<>("af-south-1", 0.0009006d),
-                    new AbstractMap.SimpleEntry<>("ap-east-1", 0.00071d),
-                    new AbstractMap.SimpleEntry<>("ap-south-1", 0.0007082d),
-                    new AbstractMap.SimpleEntry<>("ap-northeast-3", 0.0004658d),
-                    new AbstractMap.SimpleEntry<>("ap-northeast-2", 0.0004156d),
-                    new AbstractMap.SimpleEntry<>("ap-southeast-1", 0.000408d),
-                    new AbstractMap.SimpleEntry<>("ap-southeast-2", 0.00076d),
-                    new AbstractMap.SimpleEntry<>("ap-northeast-1", 0.0004658d),
-                    new AbstractMap.SimpleEntry<>("ca-central-1", 0.00012d),
-                    new AbstractMap.SimpleEntry<>("cn-north-1", 0.0005374d),
-                    new AbstractMap.SimpleEntry<>("cn-northwest-1", 0.0005374d),
-                    new AbstractMap.SimpleEntry<>("eu-central-1", 0.000311d),
-                    new AbstractMap.SimpleEntry<>("eu-west-1", 0.0002786d),
-                    new AbstractMap.SimpleEntry<>("eu-west-2", 0.000225d),
-                    new AbstractMap.SimpleEntry<>("eu-south-1", 0.0002134d),
-                    new AbstractMap.SimpleEntry<>("eu-west-3", 0.0000511d),
-                    new AbstractMap.SimpleEntry<>("eu-north-1", 0.0000088d),
-                    new AbstractMap.SimpleEntry<>("me-south-1", 0.0005059d),
-                    new AbstractMap.SimpleEntry<>("sa-east-1", 0.0000617d)
+                    Map.entry("us-east-1", 0.000379069d),
```

Comment:
Nit: With a static import of `java.util.Map.entry` you can eliminate the type qualification and make this a tad nicer to read.

Reply:
For me it is not nicer to read (more repeating characters). But maybe I misunderstand how you would apply the static import!?